lens-cube/src/main/java/org/apache/lens/cube/parse/StorageCandidate.java

```diff
@@ -137,9 +181,340 @@ public boolean contains(Candidate candidate) {
     return null;
   }
 
+  private void updatePartitionStorage(FactPartition part) throws LensException {
```

Comment:
This does not make sense. A Fact Partition will have just one storage now . In fact we can rename FactPartition to StoragePartition